### PR TITLE
[MTP] add mtp_depth_sampling

### PR DESCRIPTION
**Pull request by ZhouYuxuanYX**
Sample how many of num_nextn_predict_layers MTP depths to run each step (prefix 1..K); depths>=K skip forward, lm_head emits None, loss averages over K. Independent of reuse/share mechanism (works with mtp_reuse_last_layer aliasing and mtp_shared_last_layer).

K is drawn DETERMINISTICALLY from a private RNG seeded by a per-call counter, so every rank running the MTP layer gets the same K with NO collective. An earlier world-group broadcast(src=0) deadlocks under pp>1 (only the last stage runs the MTP layer; src=0 on first stage never joins). 'not in dict_args' guard keeps recompute idempotent.

Default off (None) => identical to before. Verified on GPU: pp=1 tests pass; pp=2 forward_backward completes for sampling on and mixed (previously hung).

### CODE REVIEW

**Comment by risemeup1111** [comment]
Open
- P2 stale skipped-depth metrics keep old values in global training logs | src/paddlefleet/models/common/language_loss/language_loss.py:673 | head 6f68cf6dc75cb6b62d914aa831132fd3b2bc88cc | clear skipped-depth keys from `logs` as well as `LanguageLoss.mtp_loss_tracker`
- P3 PR template fields still need completion | https://github.com/PaddlePaddle/PaddleFleet/pull/1550 | head 6f68cf6dc75cb6b62d914aa831132fd3b2bc88cc | fill `PR Category`, `PR Types`, and `是否引起精度变化`

File: src/paddlefleet/models/common/language_loss/language_loss.py [hunk 1]
```diff
@@ -664,6 +670,16 @@ def padding(tensor, left=False, pad_len=1):
...
 
+            # MTP depth sampling: mtp_loss holds only the computed prefix
+            # (depths 0..len-1). Drop stale tracker entries for the skipped deeper
+            # depths so the trainer does not re-log a stale value on steps where
+            # those depths were sampled out (per-depth curves stay
+            # sparse-but-correct instead of flat).
+            for _d in range(
+                len(mtp_loss), self.config.num_nextn_predict_layers
+            ):
+                LanguageLoss.mtp_loss_tracker.pop(f"mtp_{_d + 1}_loss", None)
```
Comment (risemeup1111) [inline src/paddlefleet/models/common/language_loss/language_loss.py:681]: P2 **优先级：P2**
这里清掉了 `LanguageLoss.mtp_loss_tracker` 里的跳过深度，但 `global_training_logs` 是长期复用的 dict-like 对象，后面的 `logs.update()` 不会删除已经存在的 `mtp_{n}_loss` 键。这样某一步采样出较浅 K 后，global logs 里更深层的旧 loss 仍会保留，监控/训练日志看到的曲线还是上一轮数据。建议在同一个清理逻辑里同步删除 `logs` 中被跳过的 key。

**处理要求：请针对该评论进行回复（同意并已修改请回复 Done，不同意请说明理由）。**

建议形态：
```python
logs = get_global_training_logs()
for _d in range(len(mtp_loss), self.config.num_nextn_predict_layers):
    key = f"mtp_{_d + 1}_loss"
    LanguageLoss.mtp_loss_tracker.pop(key, None)
    if logs is not None and hasattr(logs, "pop"):
        logs.pop(key, None)

if logs is not None and hasattr(logs, "update"):
    for i, loss_val in enumerate(mtp_loss):
        logs.update(**{f"mtp_{i + 1}_loss": loss_val.detach()})
```

**Review by risemeup1111 — COMMENTED** [review]
已复查，整体实现方向可以，没有需要阻塞合入的问题，细节已写在行级评论里。

- P3 **优先级：P3** 非行级：PR 模板里的 `PR Category`、`PR Types` 和 `是否引起精度变化` 仍为空，建议补齐，方便后续审核和检索。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>
